### ISSUE TITLE
Jobs randomly failing with error messages, but worker never gets called

### ISSUE THREAD

**Issue by soulchild** [taskforcesh/bullmq#1330]
We recently switched from a self-hosted Redis instance to the Azure Cache for Redis SaaS offering. 

First, everything worked as expected. A couple of days ago, the queue started behaving in weird ways: Sometimes, adding a job works flawlessly. Adding the exact same job a second time shortly after the first one failed, sporadically results in the job immediately failing. 

We added (console) logging to the worker events (`active`, `error`, etc.) and in the processing code as well, but we get absolutely no output. It looks like the job processing code doesn't run at all. What makes this even weirder is the fact that the job fails with an error message which is clearly there in our job processing code. But logging something right before throwing the error does not yield any log output. We added `job.log()` calls in our worker event handlers, but the log on the failed job is empty. 

So, to sum this up: The jobs seem to fail sporadically without even invoking the processing code and pulling some (maybe old?) error message seemingly out of thin air.

I switched back to the self-hosted Redis instance and the problem went away immediately.

I've used bullmq (always with self-hosted Redis instances) for quite some time in different projects now and never saw anything like this. 

Is it possible that the bullmq data in the Azure Redis instance somehow got corrupted or something like that? Does this ring any bell for the maintainers? 

Thank you! :pray:
